### ISSUE TITLE
Square Brackets and periods are getting removed from node names on import

### ISSUE THREAD

**Issue by timshannon** [godotengine/godot#55796]
### Godot version

v3.4.1.rc2.mono.official [d5aa00c2c]

### System information

Ubuntu 20.04.3 LTS

### Issue description

This may be intentional, but the behavior change broke some things in my project.  In an import script I parse out any values within square brackets and add it as a group on import.
```
mesh-name[group1][group2][group3]
```

Since v3.4.1.rc1 these square brackets are getting stripped out.

If I start a brand new project on 3.4.1.rc2 the issue doesn't happen.  If I use a copy of my project file from my existing project, the square brackets and periods are getting stripped out of node names, so I assume it's related to some project level setting.  I have yet to isolate which specific setting is causing the issue though.  I'll update this issue if I find it.

### Steps to reproduce

Open the attached project, import mesh.gtlf, and look at the scene.  You'll see there are no square brackets in the name.  Import that same mesh in a new clean project, and the brackets show up fine.  You can also look at the gltf file directly and see the values in the mesh name.

![image](https://user-images.githubusercontent.com/1330485/145612866-d1db7f96-9fa6-443f-a547-4877284459a0.png)

### Minimal reproduction project

[test.zip](https://github.com/godotengine/godot/files/7694316/test.zip)

**Comment by akien-mga** [comment]
This is a consequence of this bugfix: #54911. The check for `MODULE_REGEX_ENABLED` was not working due to a missing include, and this is now fixed.

https://github.com/akien-mga/godot/blob/6f4858f184870fe944cdf3cf7fdcb7eb83574d1d/modules/gltf/gltf_document.cpp#L433-L444

In 3.4-stable it would print this warning:
```
WARNING: GLTF: Legacy scene names are not supported without the RegEx module. Falling back to new names.
     at: _sanitize_scene_name (modules/gltf/gltf_document.cpp:438)
```

So it would use the new naming scheme instead of the Godot 3.2-compatible one requested by the `nodes/use_legacy_names` option for the glTF file. You can disable that option and it will use the new names which allow using square brackets.

**Comment by akien-mga** [comment]
For the record, the bug fixed by #54911 means that the `nodes/use_legacy_names` option added in 3.3 for compatibility with 3.2 probably never really worked, and thus users were forced to upgrade to the new naming scheme anyway.

#54911 fixes it but thus regresses for most users who have their pre-existing files flagged with `nodes/use_legacy_names` yet this hasn't been applied properly. So I'd suggest dropping that import option altogether in 3.4.1 and 3.5.

**Comment by akien-mga** [comment]
Duplicate of #54922
